### PR TITLE
[Whisper] Fix feature normalization in `WhisperFeatureExtractor`

### PR DESCRIPTION
**Pull request by bofenghuang**
# What does this PR do?

Hello @ArthurZucker @sanchit-gandhi  ,

In this PR, I tried to fix the feature normalization in `WhisperFeatureExtractor` which is actually not used. The line 354 takes the `input_features` from the line 340.

In addition, the `zero_mean_unit_var_norm` expects a `padded_inputs["attention_mask"]` also in the sample level (48000) as `padded_inputs["input_features"]`. IMO it should not be rescaled in the line 344 (it's also repeated in the line 363).

Please let me know what you think :)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/whisper/feature_extraction_whisper.py [hunk 1]
```diff
@@ -334,14 +334,8 @@ def __call__(
...
             pad_to_multiple_of=pad_to_multiple_of,
-            return_attention_mask=return_attention_mask,
+            return_attention_mask=return_attention_mask or do_normalize,
         )
-        # make sure list is in array format
-        input_features = padded_inputs.get("input_features").transpose(2, 0, 1)
-
-        if return_attention_mask:
-            # rescale from sample (48000) to feature (3000)
-            padded_inputs["attention_mask"] = padded_inputs["attention_mask"][:, :: self.hop_length]
```
Comment (ArthurZucker) [inline src/transformers/models/whisper/feature_extraction_whisper.py:344]: Yup! This is definetly a typo! Thanks for the catch

**Review by ArthurZucker — APPROVED** [review]
Thanks for fixing.

**Review by sgugger — APPROVED** [review]
